dvc/config.py

```diff
@@ -342,6 +348,11 @@ def edit(self, level="repo"):
         yield conf
 
         conf = self._save_paths(conf, self.files[level])
+
+        merged_conf = self._load_config_to_level(level)
```

Comment:
Looks like this loads one level too far. E.g. if level is "repo" then this will load system, global, repo and in the next line you will merge new repo config once again.

Reply:
Current level merged twice.  And I found [another pr](https://github.com/iterative/dvc/pull/3598)  would be affected by this one

Reply:
@karajan1001 Whichever PR is ready first - will be merged first :slightly_smiling_face:  So far looks like your PR is definitely ahead :slightly_smiling_face: